Gems/EditorPythonBindings/Code/Source/PythonSystemComponent.cpp

```diff
@@ -296,9 +375,9 @@ namespace EditorPythonBindings
 
     void PythonSystemComponent::Deactivate()
     {
+        StopPython(true);
```

Comment:
As long as stop python multiple times does not cause issues or over warnings this is fine. The reason this was not here in the first place was that applications not the Editor (such as the Atom Material Editor) were getting extra warnings about being shutdown too many times.

Reply:
I moved it there so the `OnPreFinalize` broadcast makes sense. The `StopPython` function makes the broadcast notification `OnPreFinalize`, anybody listening to that notification would expect that `EditorPythonEventsInterface` and `EditorPythonRunnerRequestBus` is still available, because it's pre-finalize. That could lead to bugs in the future.

The `StopPython` already is protected to being stopped twice by asking `if (!Py_IsInitialized())`, so it should be good there. And the warning issue is already taken care of by its optional boolean `silenceWarnings`.